### ISSUE TITLE
No add_project_include_directories

### ISSUE THREAD

**Issue by hughsie** [mesonbuild/meson#1139]
In most of my source files I have a:

      include_directories : [
        include_directories('..'),
      ],

... as GNOME is a heavy user of config.h.

What I wanted to have was a add_project_include_directories() function, rather than having to manually specify this for each executable. I could fall back to a `-I` argument to the compiler but this seems somewhat of a hack. Thanks!

**Comment by ignatenkobrain** [comment]
makes sense

**Comment by QuLogic** [comment]
This won't reduce redundancy, but it can shorten the files: set `toplevel_inc = include_directories('.')` in the top-level `meson.build` and then use `include_directories : [toplevel_inc]` everywhere.

**Comment by eli-schwartz** [comment]
This was implemented in #10206

```meson
incdir_dep = declare_dependency(include_directories: '..')
add_project_dependencies(incdir_dep, language: 'c')
```
